homeassistant/components/light/mysensors.py

```diff
@@ -109,118 +100,258 @@ def device_state_attributes(self):
+        """Turn the device on."""
         set_req = self.gateway.const.SetReq
-        rgb = self._rgb
-        brightness = self._brightness
-        white = self._white
 
-        if set_req.V_LIGHT in self._values and not self._state:
+        if not self._state:
             self.gateway.set_child_value(
                 self.node_id, self.child_id, set_req.V_LIGHT, 1)
 
-        if ATTR_BRIGHTNESS in kwargs and set_req.V_DIMMER in self._values and \
+        if self.gateway.optimistic:
+            # optimistically assume that light has changed state
+            self._state = True
+        super().turn_on(**kwargs)
+
+    def turn_off(self, **kwargs):
+        """Turn the device off."""
+        set_req = self.gateway.const.SetReq
+        value_type = kwargs.get('value_type')
+        value = kwargs.get('value')
+        value_type = (
+            set_req.V_LIGHT
+            if set_req.V_LIGHT in self._values else value_type)
+        value = 0 if set_req.V_LIGHT in self._values else value
+        super().turn_off(value_type=value_type, value=value)
+
+    def update(self):
+        """Update the controller with the latest value from a sensor."""
+        super().update()
+        value_type = self.gateway.const.SetReq.V_LIGHT
+        if value_type in self._values:
+            self._values[value_type] = (
+                STATE_ON if int(self._values[value_type]) == 1 else STATE_OFF)
+            self._state = self._values[value_type] == STATE_ON
+
+
+class MySensorsLightDimmer(MySensorsLightLight):
+    """Dimmer child class to MySensorsLight."""
+
+    def __init__(self, *args):
+        """Setup instance attributes."""
+        self._brightness = None
+        super().__init__(*args)
+
+    @property
+    def brightness(self):
+        """Brightness of this light between 0..255."""
+        return self._brightness
+
+    def turn_on(self, **kwargs):
+        """Turn the device on."""
+        set_req = self.gateway.const.SetReq
+        brightness = self._brightness
+
+        if ATTR_BRIGHTNESS in kwargs and \
                 kwargs[ATTR_BRIGHTNESS] != self._brightness:
             brightness = kwargs[ATTR_BRIGHTNESS]
             percent = round(100 * brightness / 255)
             self.gateway.set_child_value(
                 self.node_id, self.child_id, set_req.V_DIMMER, percent)
 
+        if self.gateway.optimistic:
+            # optimistically assume that light has changed state
+            self._brightness = brightness
+        super().turn_on(**kwargs)
+
+    def turn_off(self, **kwargs):
+        """Turn the device off."""
+        set_req = self.gateway.const.SetReq
+        value_type = kwargs.get('value_type')
+        value = kwargs.get('value')
+        value_type = (
+            set_req.V_DIMMER
+            if set_req.V_DIMMER in self._values else value_type)
+        value = 0 if set_req.V_DIMMER in self._values else value
+        super().turn_off(value_type=value_type, value=value)
```

Comment:
I do not feel like this is the correct way of using `super`. I would suggest you rename `turn_off` in the base class to be `_turn_off` and have it be a helper method to this turn off (because that's what it is).